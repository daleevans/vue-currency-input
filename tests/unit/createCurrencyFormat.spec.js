import createCurrencyFormat from '../../src/utils/createCurrencyFormat'

describe('createCurrencyFormat', () => {
  describe('i18n', () => {
    it('de-DE_EUR', () => {
      expect(createCurrencyFormat({ locale: 'de-DE', currency: 'EUR' })).toMatchSnapshot()
    })

    it('de-CH_EUR', () => {
      expect(createCurrencyFormat({ locale: 'de-CH', currency: 'USD' })).toMatchSnapshot()
    })

    it('es-ES_EUR', () => {
      expect(createCurrencyFormat({ locale: 'es-ES', currency: 'EUR' })).toMatchSnapshot()
    })

    it('nl-NL_EUR', () => {
      expect(createCurrencyFormat({ locale: 'nl-NL', currency: 'EUR' })).toMatchSnapshot()
    })

    it('en-US_USD', () => {
      expect(createCurrencyFormat({ locale: 'en-US', currency: 'USD' })).toMatchSnapshot()
    })

    it('zh_CNY', () => {
      expect(createCurrencyFormat({ locale: 'zh', currency: 'CNY' })).toMatchSnapshot()
    })

    it('en-GB_GBP', () => {
      expect(createCurrencyFormat({ locale: 'en-GB', currency: 'GBP' })).toMatchSnapshot()
    })

    it('en-IN_INR', () => {
      expect(createCurrencyFormat({ locale: 'en-GB', currency: 'INR' })).toMatchSnapshot()
    })

    it('pt_BRL', () => {
      expect(createCurrencyFormat({ locale: 'pt', currency: 'BRL' })).toMatchSnapshot()
    })

    it('ja_JPY', () => {
      expect(createCurrencyFormat({ locale: 'ja', currency: 'JPY' })).toMatchSnapshot()
    })

    it('ar-SA_SAR', () => {
      expect(createCurrencyFormat({ locale: 'ar-SA', currency: 'SAR' })).toMatchSnapshot()
    })
  })

  it('custom prefix/suffix', () => {
    expect(createCurrencyFormat({ locale: 'de-DE', currency: { prefix: '₿ ' } })).toMatchSnapshot()
    expect(createCurrencyFormat({ locale: 'de-DE', currency: { suffix: ' Euro' } })).toMatchSnapshot()
    expect(createCurrencyFormat({ locale: 'de-DE', currency: null })).toMatchSnapshot()
  })

  it('custom precision', () => {
    expect(createCurrencyFormat({ locale: 'de-DE', currency: 'EUR', precision: 0 })).toMatchSnapshot()
    expect(createCurrencyFormat({ locale: 'de-DE', currency: 'EUR', precision: { min: 0, max: 0 } })).toMatchSnapshot()
    expect(createCurrencyFormat({ locale: 'de-DE', currency: null, precision: { min: 0, max: 2 } })).toMatchSnapshot()
    expect(createCurrencyFormat({ locale: 'de-DE', currency: { suffix: ' €' }, precision: { min: 0, max: 2 } })).toMatchSnapshot()
    expect(createCurrencyFormat({ locale: 'ja', currency: 'JPY', precision: { min: 2, max: 2 } })).toMatchSnapshot()
  })
})
