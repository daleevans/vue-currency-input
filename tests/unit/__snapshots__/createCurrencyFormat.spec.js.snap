// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createCurrencyFormat custom precision 1`] = `
Object {
  "decimalSymbol": ",",
  "digits": Array [
    "0",
    "1",
    "2",
    "3",
    "4",
    "5",
    "6",
    "7",
    "8",
    "9",
  ],
  "groupingSymbol": ".",
  "maximumFractionDigits": 0,
  "minimumFractionDigits": 0,
  "minusSymbol": "-",
  "negativePrefix": "-",
  "prefix": "",
  "suffix": " €",
}
`;

exports[`createCurrencyFormat custom precision 2`] = `
Object {
  "decimalSymbol": ",",
  "digits": Array [
    "0",
    "1",
    "2",
    "3",
    "4",
    "5",
    "6",
    "7",
    "8",
    "9",
  ],
  "groupingSymbol": ".",
  "maximumFractionDigits": 0,
  "minimumFractionDigits": 0,
  "minusSymbol": "-",
  "negativePrefix": "-",
  "prefix": "",
  "suffix": " €",
}
`;

exports[`createCurrencyFormat custom precision 3`] = `
Object {
  "decimalSymbol": ",",
  "digits": Array [
    "0",
    "1",
    "2",
    "3",
    "4",
    "5",
    "6",
    "7",
    "8",
    "9",
  ],
  "groupingSymbol": ".",
  "maximumFractionDigits": 2,
  "minimumFractionDigits": 0,
  "minusSymbol": "-",
  "negativePrefix": "-",
  "prefix": "",
  "suffix": "",
}
`;

exports[`createCurrencyFormat custom precision 4`] = `
Object {
  "decimalSymbol": ",",
  "digits": Array [
    "0",
    "1",
    "2",
    "3",
    "4",
    "5",
    "6",
    "7",
    "8",
    "9",
  ],
  "groupingSymbol": ".",
  "maximumFractionDigits": 2,
  "minimumFractionDigits": 0,
  "minusSymbol": "-",
  "negativePrefix": "-",
  "prefix": "",
  "suffix": " €",
}
`;

exports[`createCurrencyFormat custom precision 5`] = `
Object {
  "decimalSymbol": undefined,
  "digits": Array [
    "0",
    "1",
    "2",
    "3",
    "4",
    "5",
    "6",
    "7",
    "8",
    "9",
  ],
  "groupingSymbol": ",",
  "maximumFractionDigits": 0,
  "minimumFractionDigits": 0,
  "minusSymbol": "-",
  "negativePrefix": "-￥",
  "prefix": "￥",
  "suffix": "",
}
`;

exports[`createCurrencyFormat custom prefix/suffix 1`] = `
Object {
  "decimalSymbol": ",",
  "digits": Array [
    "0",
    "1",
    "2",
    "3",
    "4",
    "5",
    "6",
    "7",
    "8",
    "9",
  ],
  "groupingSymbol": ".",
  "maximumFractionDigits": 2,
  "minimumFractionDigits": 2,
  "minusSymbol": "-",
  "negativePrefix": "-₿ ",
  "prefix": "₿ ",
  "suffix": "",
}
`;

exports[`createCurrencyFormat custom prefix/suffix 2`] = `
Object {
  "decimalSymbol": ",",
  "digits": Array [
    "0",
    "1",
    "2",
    "3",
    "4",
    "5",
    "6",
    "7",
    "8",
    "9",
  ],
  "groupingSymbol": ".",
  "maximumFractionDigits": 2,
  "minimumFractionDigits": 2,
  "minusSymbol": "-",
  "negativePrefix": "-",
  "prefix": "",
  "suffix": " Euro",
}
`;

exports[`createCurrencyFormat custom prefix/suffix 3`] = `
Object {
  "decimalSymbol": ",",
  "digits": Array [
    "0",
    "1",
    "2",
    "3",
    "4",
    "5",
    "6",
    "7",
    "8",
    "9",
  ],
  "groupingSymbol": ".",
  "maximumFractionDigits": 2,
  "minimumFractionDigits": 2,
  "minusSymbol": "-",
  "negativePrefix": "-",
  "prefix": "",
  "suffix": "",
}
`;

exports[`createCurrencyFormat i18n ar-SA_SAR 1`] = `
Object {
  "decimalSymbol": "٫",
  "digits": Array [
    "٠",
    "١",
    "٢",
    "٣",
    "٤",
    "٥",
    "٦",
    "٧",
    "٨",
    "٩",
  ],
  "groupingSymbol": "٬",
  "maximumFractionDigits": 2,
  "minimumFractionDigits": 2,
  "minusSymbol": "-",
  "negativePrefix": "؜-",
  "prefix": "",
  "suffix": " ر.س.‏",
}
`;

exports[`createCurrencyFormat i18n de-CH_EUR 1`] = `
Object {
  "decimalSymbol": ".",
  "digits": Array [
    "0",
    "1",
    "2",
    "3",
    "4",
    "5",
    "6",
    "7",
    "8",
    "9",
  ],
  "groupingSymbol": "’",
  "maximumFractionDigits": 2,
  "minimumFractionDigits": 2,
  "minusSymbol": "-",
  "negativePrefix": "$-",
  "prefix": "$ ",
  "suffix": "",
}
`;

exports[`createCurrencyFormat i18n de-DE_EUR 1`] = `
Object {
  "decimalSymbol": ",",
  "digits": Array [
    "0",
    "1",
    "2",
    "3",
    "4",
    "5",
    "6",
    "7",
    "8",
    "9",
  ],
  "groupingSymbol": ".",
  "maximumFractionDigits": 2,
  "minimumFractionDigits": 2,
  "minusSymbol": "-",
  "negativePrefix": "-",
  "prefix": "",
  "suffix": " €",
}
`;

exports[`createCurrencyFormat i18n en-GB_GBP 1`] = `
Object {
  "decimalSymbol": ".",
  "digits": Array [
    "0",
    "1",
    "2",
    "3",
    "4",
    "5",
    "6",
    "7",
    "8",
    "9",
  ],
  "groupingSymbol": ",",
  "maximumFractionDigits": 2,
  "minimumFractionDigits": 2,
  "minusSymbol": "-",
  "negativePrefix": "-£",
  "prefix": "£",
  "suffix": "",
}
`;

exports[`createCurrencyFormat i18n en-IN_INR 1`] = `
Object {
  "decimalSymbol": ".",
  "digits": Array [
    "0",
    "1",
    "2",
    "3",
    "4",
    "5",
    "6",
    "7",
    "8",
    "9",
  ],
  "groupingSymbol": ",",
  "maximumFractionDigits": 2,
  "minimumFractionDigits": 2,
  "minusSymbol": "-",
  "negativePrefix": "-₹",
  "prefix": "₹",
  "suffix": "",
}
`;

exports[`createCurrencyFormat i18n en-US_USD 1`] = `
Object {
  "decimalSymbol": ".",
  "digits": Array [
    "0",
    "1",
    "2",
    "3",
    "4",
    "5",
    "6",
    "7",
    "8",
    "9",
  ],
  "groupingSymbol": ",",
  "maximumFractionDigits": 2,
  "minimumFractionDigits": 2,
  "minusSymbol": "-",
  "negativePrefix": "-$",
  "prefix": "$",
  "suffix": "",
}
`;

exports[`createCurrencyFormat i18n es-ES_EUR 1`] = `
Object {
  "decimalSymbol": ",",
  "digits": Array [
    "0",
    "1",
    "2",
    "3",
    "4",
    "5",
    "6",
    "7",
    "8",
    "9",
  ],
  "groupingSymbol": ".",
  "maximumFractionDigits": 2,
  "minimumFractionDigits": 2,
  "minusSymbol": "-",
  "negativePrefix": "-",
  "prefix": "",
  "suffix": " €",
}
`;

exports[`createCurrencyFormat i18n ja_JPY 1`] = `
Object {
  "decimalSymbol": undefined,
  "digits": Array [
    "0",
    "1",
    "2",
    "3",
    "4",
    "5",
    "6",
    "7",
    "8",
    "9",
  ],
  "groupingSymbol": ",",
  "maximumFractionDigits": 0,
  "minimumFractionDigits": 0,
  "minusSymbol": "-",
  "negativePrefix": "-￥",
  "prefix": "￥",
  "suffix": "",
}
`;

exports[`createCurrencyFormat i18n nl-NL_EUR 1`] = `
Object {
  "decimalSymbol": ",",
  "digits": Array [
    "0",
    "1",
    "2",
    "3",
    "4",
    "5",
    "6",
    "7",
    "8",
    "9",
  ],
  "groupingSymbol": ".",
  "maximumFractionDigits": 2,
  "minimumFractionDigits": 2,
  "minusSymbol": "-",
  "negativePrefix": "€ -",
  "prefix": "€ ",
  "suffix": "",
}
`;

exports[`createCurrencyFormat i18n pt_BRL 1`] = `
Object {
  "decimalSymbol": ",",
  "digits": Array [
    "0",
    "1",
    "2",
    "3",
    "4",
    "5",
    "6",
    "7",
    "8",
    "9",
  ],
  "groupingSymbol": ".",
  "maximumFractionDigits": 2,
  "minimumFractionDigits": 2,
  "minusSymbol": "-",
  "negativePrefix": "-R$ ",
  "prefix": "R$ ",
  "suffix": "",
}
`;

exports[`createCurrencyFormat i18n zh_CNY 1`] = `
Object {
  "decimalSymbol": ".",
  "digits": Array [
    "0",
    "1",
    "2",
    "3",
    "4",
    "5",
    "6",
    "7",
    "8",
    "9",
  ],
  "groupingSymbol": ",",
  "maximumFractionDigits": 2,
  "minimumFractionDigits": 2,
  "minusSymbol": "-",
  "negativePrefix": "-¥",
  "prefix": "¥",
  "suffix": "",
}
`;
